### ISSUE TITLE
Consistency in Quantity/Range options

### ISSUE THREAD

**Issue by BSnoeijerCD** [cdisc-org/DDF-RA#523]
For the substance.strength numerator, the user can optionally choose to specify a range (using the range class) or single value (using the quantity class).
This option is not available for plannedAge, plannedEnrollmentNumber and plannedCompletionNumber. However, planned completion number and planned enrollment number are likely to be a single value. Both minValue and maxValue are required though. 

To be consistent with the substance.strength approach we propose to add a similar structure for plannedEnrollmentNumber and plannedCompletion number (e.g. either allow range or quantity class to be used). 

In summary:
- plannedEnrollmentNumber, plannedCompletionNumber: allow single value
- plannedAge: allow for a different unit for min and max
- complex data type: allow for specifying both as a complex data type in the UML

| Item  | Ticket |
| ------------- | ------------- |
| Model |  #552 |
| CT |  #553 |
| API | #554 |
| Rules | #556 |
| IG | #555 |
| Test Data | #557  |

**Comment by dih-cdisc** [comment]
@BSnoeijerCD Check is this still needed or have change been made?

**Comment by BSnoeijerCD** [comment]
Also consider to split min and max age into two different values again since the unit for min age can be different from that of max age (Examples "3 Days to 6 Months" or "6 Months to 6 years" ranges).

**Comment by BSnoeijerCD** [comment]
This was discussed during Scrum today. UML pictures below picture all the current usage of the quantity and range class.
@dih-cdisc @ASL-rmarshall Resolving the issue of separate minvalue and maxvalue unit in the age range can be done by pointing the minvalue and maxvalue to the quantity class and removing the unit relationship from the range class. See first picture below. This also requires users to specify the unit twice for other ranges in the model:
![USDM_quantityRange_Suggestion1](https://github.com/user-attachments/assets/815a6c7b-7dc7-476b-99d2-8d972b1e4f4c)

Allowing for a single or a range can be addressed by pointing to a Range or a Quantity class. However, if we want to point to these classes as complex datatypes we cannot specify it as such in the UML as we cannot point to two different data types for one attribute. To address this, an overarching datatype can help. In the example below this overarching complex datatype is now called Amount which is then inherited by both range and quantity classes. Both plannedEnrollmentNumber and plannedCompletionNumber as well as the strength numerator can point to this new class leaving the user the choice. (like we did for the StudyArm.population relationship). If there is no choice the model will specifically point to the quantity or range class. Note that in this case the amount class is just a container class solely for this purpose. Reuse of unit or other attributes is not feasible if we want to split it in a minvalue with unit and a maxvalue with unit. If we combine it with the suggestion above the range class will point to the quantity class.
![USDM_quantityRange_Suggestion2](https://github.com/user-attachments/assets/fe3b4235-c241-47be-98ba-9f47a226acd3)

**Comment by ASL-rmarshall** [comment]
@BSnoeijerCD @dih-cdisc A possible option to allow flexibility would be to:
- Use the `Amount` general class with an optional `unit`. 
- Define `minValue` and `maxValue` with a "data type" of `Quantity`.

![image](https://github.com/user-attachments/assets/b1320a6c-0ea0-4157-a763-3f3b21dfd553)

This means that the unit could be specified either for the whole range or for the individual min/max values.  However, you'd probably then need a rules to ensure that a unit is not specified for both the whole range and min/max values.

**Comment by BSnoeijerCD** [comment]
@EMuhlbradt @czwickl  
- all new attributes should be complex datatype relationships (like the numeratorQuantity and numeratorRange)
- the definitions for plannedEnrollmentNumberQuantity and plannedCompletionNumberQuantity include a reference to range instead of quantity.

**Comment by ASL-rmarshall** [comment]
@EMuhlbradt @czwickl 
I agree with Berber's findings and I have no additional ones of my own - just a question: if the new quantity and range attributes for enrollment number and completion number are changed to complex data types, does this mean that:
- Their definitions need to be re-styled (maybe back to being similar to the original definitions for `plannedEnrollmentNumber` and `plannedCompletionNumber`)?
- They each need an NCI Code and Preferred Term (again possibly similar to the original values for `plannedEnrollmentNumber` and `plannedCompletionNumber`)?

**Comment by EMuhlbradt** [comment]
@ASL-rmarshall @BSnoeijerCD @dih-cdisc @czwickl : Attached is the updated CT draft file for this ticket. The change in here is that 12 items that were previously defined as attributes are now defined as complex datatype relationships with NCI C-codes, PTs, and concept definitions.
[523.xlsx](https://github.com/user-attachments/files/18410390/523.xlsx)

**Comment by BSnoeijerCD** [comment]
> @ASL-rmarshall @BSnoeijerCD @dih-cdisc @czwickl : Attached is the updated CT draft file for this ticket. The change in here is that 12 items that were previously defined as attributes are now defined as complex datatype relationships with NCI C-codes, PTs, and concept definitions. [523.xlsx](https://github.com/user-attachments/files/18410390/523.xlsx)

@EMuhlbradt: Checked and ok now

**Comment by ASL-rmarshall** [comment]
@EMuhlbradt @czwickl @BSnoeijerCD 
I've reviewed the updated spreadsheet - it looks good, except that I suspect that the "N/A" values in columns C and D on the 523_new sheet need to be replaced according to class with the appropriate attribute codelist:
| Codelist Code | Codelist Name                                     |
| ------------- | ------------------------------------------------- |
| C207432       | DDF Population Definition Attribute Terminology   |
| C207439       | DDF Study Cohort Attribute Terminology            |
| C188706       | DDF Study Design Population Attribute Terminology |

**Comment by EMuhlbradt** [comment]
@ASL-rmarshall @BSnoeijerCD @dih-cdisc @czwickl : Updates in 3-11-0 release in github ready for review.

**Comment by BSnoeijerCD** [comment]
@EMuhlbradt @czwickl 
Thank you. All changes are well incorporated. 
While checking I found some inconsistent complex datatype relationships:
- strength denominator should be a complex datatype relationship as it goes to the quantity class
- abbreviatedText and expandedText are indicated as complex datatype relationships while they are just attributes.
Is it possible to update that?

**Comment by EMuhlbradt** [comment]
> [@EMuhlbradt](https://github.com/EMuhlbradt) [@czwickl](https://github.com/czwickl) Thank you. All changes are well incorporated. While checking I found some inconsistent complex datatype relationships:
> 
> * strength denominator should be a complex datatype relationship as it goes to the quantity class
> * abbreviatedText and expandedText are indicated as complex datatype relationships while they are just attributes.
>   Is it possible to update that?

@BSnoeijerCD updates above made in 3-11-0 release on github

**Comment by BSnoeijerCD** [comment]
> > [@EMuhlbradt](https://github.com/EMuhlbradt) [@czwickl](https://github.com/czwickl) Thank you. All changes are well incorporated. While checking I found some inconsistent complex datatype relationships:
> > 
> > * strength denominator should be a complex datatype relationship as it goes to the quantity class
> > * abbreviatedText and expandedText are indicated as complex datatype relationships while they are just attributes.
> >   Is it possible to update that?
> 
> [@BSnoeijerCD](https://github.com/BSnoeijerCD) updates above made in 3-11-0 release on github

@EMuhlbradt: Could you change the role for abbreviatedText and expandedText to "Attribute"? They now have the role "Relationship in the spreadsheet on github. The strength.denominator is ok now.

**Comment by BSnoeijerCD** [comment]
> > > [@EMuhlbradt](https://github.com/EMuhlbradt) [@czwickl](https://github.com/czwickl) Thank you. All changes are well incorporated. While checking I found some inconsistent complex datatype relationships:
> > > 
> > > * strength denominator should be a complex datatype relationship as it goes to the quantity class
> > > * abbreviatedText and expandedText are indicated as complex datatype relationships while they are just attributes.
> > >   Is it possible to update that?
> > 
> > 
> > [@BSnoeijerCD](https://github.com/BSnoeijerCD) updates above made in 3-11-0 release on github
> 
> [@EMuhlbradt](https://github.com/EMuhlbradt): Could you change the role for abbreviatedText and expandedText to "Attribute"? They now have the role "Relationship in the spreadsheet on github. The strength.denominator is ok now.

Everything Ok now. So moved to Done.
